src/firefly/js/templates/lightcurve/LcPhaseTable.js

```diff
@@ -0,0 +1,232 @@
+ *          to be after or replace certain column
+ * @param {TableModel} tbl
+ * @param {array|string} srcCols
+ * @param {number|string} targetCol
+ * @param {number} position 1: after the target columm, -1: before the target column, 0: at the target column
+ */
+function locateTableColumns(tbl, srcCols, targetCol, position = 1) {
+    var {columns, data} = tbl.tableData;
+
+    if (!isArray(srcCols)) {
+        srcCols = [srcCols];
+    }
+
+    var colIdx = [];
+    var cols = srcCols.reduce((prev, srcCol) => {
+            var idx = columns.findIndex((col) => (col.name === srcCol));
+
+            if (idx >= 0) {
+                prev.push(columns[idx]);
+                columns.splice(idx, 1);
+                colIdx.push(idx);
+            }
+            return prev;
+        }, []);
+
+
+    var targetIdx = isString(targetCol) ? columns.findIndex((col) => (col.name === targetCol)) : targetCol;
+    if (targetIdx < 0) {
+        targetIdx = 0;
+        position = 1;
+    }
+
+    if (targetIdx >= 0) {
+        var deleteCount;
+
+        if (position > 0) {
+            targetIdx += 1;
+            deleteCount = 0;
+        } else if (position < 0) {
+            deleteCount = 0;
+        } else {
+            deleteCount = 1;
+        }
+
+        columns.splice( targetIdx, deleteCount,...cols);
+
+        data.forEach((dataRow) => {
+                var content = colIdx.reduce((prev, idx) => {
+                    prev.push(dataRow[idx]);
+                    dataRow.splice(idx, 1);
+                    return prev;
+                }, []);
+                dataRow.splice(targetIdx, deleteCount,...content);
+            });
+
+        set(tbl.tableData, ['columns'], columns);
+        set(tbl.tableData, ['data'], data);
+    }
+
+}
+
+/**
+ * @summary create table request object for phase folded table
+ * @param {string} period period in string
+ * @param {string} time  time column
+ * @param {string} flux flux name
+ * @param {TableModel} tbl
+ * @returns {TableRequest}
+ */
+function getPhaseFoldingRequest(period, time, flux, tbl) {
+    const cutoutSize = getValidValueFrom(FieldGroupUtils.getGroupFields(LC.FG_VIEWER_FINDER), 'cutoutSize');
+
+    return makeTblRequest('PhaseFoldedProcessor', LC.PHASE_FOLDED, {
+        period_days: period,
+        table_name: 'folded_table',
+        cutout_size: cutoutSize ? cutoutSize : undefined,
+        flux,
+        x: time,
+        original_table: tbl.tableMeta.tblFilePath
+    },  {tbl_id:LC.PHASE_FOLDED});
+
+}
+
+/**
+ * @summary duplicate the phase cycle to the phase folded table
+ * @param {string} timeCol
+ * @param {number} period
+ * @param {TableModel} phaseTable
+ */
+function repeatDataCycle(timeCol, period, phaseTable) {
+    var tIdx = phaseTable.tableData.columns.findIndex((c) => (c.name === timeCol));
+    var fIdx = phaseTable.tableData.columns.findIndex((c) => (c.name === LC.PHASE_CNAME));
+    var {totalRows, tbl_id, title} = phaseTable;
```

Comment:
Code might be a little cleaner if you also pull `tableData` our of `phaseTable`  and moved this  line to the beginning of the function so that did not have to keep repeating `phaseTable.tableData`

```
var {totalRows, tbl_id, title, tableData} = phaseTable;
```